### PR TITLE
converts some attackby to toolact

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
fix: converted some attackby to toolact
/:cl:

- [x] crate

- [x] swarmer

- [x] echair

- [ ] bed(needs testing)

- [x] tiles

- [ ] dresser (needs testing

### CODE REVIEW

File: code/game/objects/items/stacks/tiles/light.dm [hunk 1]
```diff
@@ -19,8 +19,7 @@
 	else
 		state = 0 //fine
 
-/obj/item/stack/tile/light/attackby(obj/item/O, mob/user, params)
-	if(istype(O, /obj/item/crowbar))
+/obj/item/stack/tile/light/crowbar_act(mob/living/user, obj/item/I)
 		new/obj/item/stack/sheet/metal(user.loc)
 		amount--
 		new/obj/item/stack/light_w(user.loc)
```
Comment (XDTM) [inline code/game/objects/items/stacks/tiles/light.dm]: there's an `else return ..()` to remove below this

**Comment by ExcessiveUseOfCobblestone** [comment]
CL not needed unless you're changing functionality

File: code/game/objects/items/stacks/tiles/light.dm [hunk 2]
```diff
@@ -19,12 +19,10 @@
...
-	if(istype(O, /obj/item/crowbar))
-		new/obj/item/stack/sheet/metal(user.loc)
-		amount--
-		new/obj/item/stack/light_w(user.loc)
-		if(amount <= 0)
-			qdel(src)
-	else
-		return ..()
+/obj/item/stack/tile/light/crowbar_act(mob/living/user, obj/item/I)
+	new/obj/item/stack/sheet/metal(user.loc)
```
Comment (ShizCalev) [inline code/game/objects/items/stacks/tiles/light.dm]: drop_location

File: code/game/objects/items/stacks/tiles/light.dm [hunk 3]
```diff
@@ -19,12 +19,10 @@
...
+	amount--
+	new/obj/item/stack/light_w(user.loc)
```
Comment (ShizCalev) [inline code/game/objects/items/stacks/tiles/light.dm]: drop_location

File: code/game/objects/structures/crates_lockers/crates/large.dm [hunk 4]
```diff
@@ -1,43 +1,38 @@
...
-		for(var/i in 1 to material_drop_amount)
-			new material_drop(src)
-		for(var/atom/movable/AM in contents)
-			AM.forceMove(T)
+/obj/structure/closet/crate/large/crowbar_act(mob/living/user, obj/item/I)
+	if(manifest)
+		tear_manifest(user)
 
-		qdel(src)
+	user.visible_message("[user] pries \the [src] open.", \
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/crates/large.dm]: remove \the

File: code/game/objects/structures/crates_lockers/crates/large.dm [hunk 5]
```diff
@@ -1,43 +1,38 @@
...
+					 "<span class='notice'>You pry open \the [src].</span>", \
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/crates/large.dm]: \the

File: code/game/objects/structures/crates_lockers/crates/large.dm [hunk 6]
```diff
@@ -1,43 +1,38 @@
...
+					 "<span class='italics'>You hear splitting wood.</span>")
+	playsound(src.loc, 'sound/weapons/slashmiss.ogg', 75, 1)
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/crates/large.dm]: just src, replace the 1 with TRUE

File: code/game/objects/structures/crates_lockers/crates/large.dm [hunk 7]
```diff
@@ -1,43 +1,38 @@
...
 
-	else
-		if(user.a_intent == INTENT_HARM)	//Only return  ..() if intent is harm, otherwise return 0 or just end it.
-			return ..()						//Stops it from opening and turning invisible when items are used on it.
+	var/turf/T = get_turf(src)
+	for(var/i in 1 to material_drop_amount)
+		new material_drop(src)
+	for(var/atom/movable/AM in contents)
+		AM.forceMove(T)
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/crates/large.dm]: forcemove to drop_location instead

File: code/game/objects/structures/crates_lockers/crates/large.dm [hunk 8]
```diff
@@ -1,43 +1,38 @@
...
-		qdel(src)
+	user.visible_message("[user] pries \the [src] open.", \
+					 "<span class='notice'>You pry open \the [src].</span>", \
+					 "<span class='italics'>You hear splitting wood.</span>")
+	playsound(src.loc, 'sound/weapons/slashmiss.ogg', 75, 1)
 
-	else
-		if(user.a_intent == INTENT_HARM)	//Only return  ..() if intent is harm, otherwise return 0 or just end it.
-			return ..()						//Stops it from opening and turning invisible when items are used on it.
+	var/turf/T = get_turf(src)
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/crates/large.dm]: can remove this after changing it to droploc

File: code/game/objects/structures/electricchair.dm [hunk 9]
```diff
@@ -10,15 +10,15 @@
...
-	if(istype(W, /obj/item/wrench))
-		var/obj/structure/chair/C = new /obj/structure/chair(loc)
-		W.play_tool_sound(src)
-		C.setDir(dir)
-		part.forceMove(loc)
-		part.master = null
-		part = null
-		qdel(src)
+/obj/structure/chair/e_chair/wrench_act(mob/living/user, obj/item/I)
+	var/obj/structure/chair/C = new /obj/structure/chair(loc)
```
Comment (ShizCalev) [inline code/game/objects/structures/electricchair.dm]: var/obj/structure/chair/C = new(loc)

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/swarmer/swarmer.dm [hunk 10]
```diff
@@ -40,15 +40,14 @@
...
-			"<span class='italics'>You hear clicking.</span>")
-		new /obj/item/deactivated_swarmer(get_turf(src))
-		qdel(src)
-	else
-		..()
+/obj/effect/mob_spawn/swarmer/screwdriver_act(mob/living/user, obj/item/I)
+	user.visible_message("<span class='warning'>[usr.name] deactivates [src].</span>",
+		"<span class='notice'>After some fiddling, you find a way to disable [src]'s power source.</span>",
+		"<span class='italics'>You hear clicking.</span>")
+	new /obj/item/deactivated_swarmer(get_turf(src))
```
Comment (ShizCalev) [inline code/modules/antagonists/swarmer/swarmer.dm]: drop_location

**Pull request closed without merging**